cpp/src/fil/infer.cu

```diff
@@ -23,23 +23,27 @@ namespace fil {
 using namespace MLCommon;
 
 // vec wraps float[N] for cub::BlockReduce
-template <int N>
+template <int N, typename T>
```

Comment:
I don't think `T` other than `float` is actually used in this pull request.

Reply:
unsigned int is used for INT_CLASS_LABEL